### ISSUE TITLE
When instance was created: __construct is suggested

### ISSUE THREAD

**Issue by jens1o** [felixfbecker/php-language-server#314]
![image](https://cloud.githubusercontent.com/assets/11234139/23302599/985dfb46-fa8f-11e6-8f6b-5a1f6c642a6a.png)
Maybe exclude all magic methods from being suggested?

**Comment by felixfbecker** [comment]
Well, technically you can actually call them. Just like you can call `__toString()`. Whether that is a good practice is a different story. It is of course annoying that these are always on the top because underscore sorts before letters

**Comment by jens1o** [comment]
Could we introduce a setting that checks for methods that won't be suggested, just like this:
```js
{
     php.intellisense.excludeMethods: [
         '__construct',
         '__call',
         '__clone',
         '__sleep',
         '__wake'
    ]
}
```

**Comment by felixfbecker** [comment]
Do you think it's beneficial for a user to configure these?

**Comment by jens1o** [comment]
Hmm, you said (in a indirect way) that there are users who call them directly, so here we are. Didn't you?

**Comment by felixfbecker** [comment]
No, I said that you can _technically_ call them. I think it would be fine to filter out the magic methods as they are not intended to be called directly by their nature. A simple filter for a `__` prefix would do.

**Comment by jens1o** [comment]
Hmm, I don't think so(regarding the filter). Because there are open-source libs that calls a method with `__` in front of it. See: https://github.com/WoltLab/WCF/blob/master/wcfsetup/install/files/lib/page/IPage.class.php#L16

**Comment by felixfbecker** [comment]
Then these libs are wrong, methods with double underscores are reserved. http://php.net/manual/en/language.oop5.magic.php

> PHP reserves all function names starting with __ as magical. It is recommended that you do not use function names with __ in PHP unless you want some documented magic functionality.
